lib/fireblocks/jsonhttp.go

```diff
@@ -0,0 +1,98 @@
+package fireblocks
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+)
+
+// jsonHTTP provides a simple interface for sending JSON HTTP requests.
+type jsonHTTP struct {
+	apiKey       string
+	clientSecret string
+	host         string
+	http         http.Client
+}
+
+// newJSONHTTP creates a new jsonHTTP.
+func newJSONHTTP(host string, apiKey string, clientSecret string) jsonHTTP {
+	return jsonHTTP{
+		host:         host,
+		apiKey:       apiKey,
+		clientSecret: clientSecret,
+	}
+}
+
+// Send sends an JSON HTTP request with the json formatted request as body.
+// It marshals the response body into the provided response pointer if not nil.
+func (c jsonHTTP) Send(ctx context.Context, endpoint string, httpMethod string, request any, headers map[string]string, response any) error {
```

Comment:
pass in response object to unmarshal response body into instead of returning bytes